Makefile-libostree.am

```diff
@@ -266,10 +266,11 @@ if BUILDOPT_INTROSPECTION
 OSTree-1.0.gir: libostree-1.la Makefile
 OSTree_1_0_gir_EXPORT_PACKAGES = ostree-1
 OSTree_1_0_gir_INCLUDES = Gio-2.0
+OSTree_1_0_gir_C_INCLUDES = ostree.h
```

Comment:
That's certainly a lot cleaner. Do you know if there's a certain version of gobject-introspection where this was introduced? Also, does it recurse through the included header? `ostree.h` is just a meta header that includes all the other real headers.

Reply:
This variable has been added in GObject introspection 1.51.5 https://gitlab.gnome.org/GNOME/gobject-introspection/-/commit/76ea47cfff778a215e982d86b6c6458bcf69354b
It doesn't change much of the result and it only useful for the consumers (like in Vala that need to generate C code)

Reply:
That's pretty old, but currently configure.ac specifies `GOBJECT_INTROSPECTION_CHECK([1.34.0])`. Quick distro check:

[Debian](https://packages.debian.org/search?keywords=gobject-introspection&searchon=sourcenames&suite=all&section=all) - buster (old stable) has 1.58.3
[Fedora](https://src.fedoraproject.org/rpms/gobject-introspection) - F36 has 1.72.1
[Ubuntu](https://packages.ubuntu.com/search?keywords=gobject-introspection&searchon=sourcenames&suite=all&section=all) - bionic (18.04) has 1.56.1
[CentOS](https://www.rpmfind.net/linux/rpm2html/search.php?query=gobject-introspection&submit=Search+...&system=centos&arch=) and presumably RHEL - 7 has 1.56.1

So, I think it's fair to assume 1.51.5 on even the oldest LTS distros, but I would bump that version in configure.ac.

Reply:
Could you attach the generated `OSTree-1.0.gir` before and after the change? Or just a diff of them?

Reply:
<details><summary>Diff</summary>
<p>

```diff
*** OSTree-1.0.gir.old	2023-02-08 16:52:04.601262975 +0100
--- OSTree-1.0.gir.new	2023-02-08 16:57:33.992850211 +0100
***************
*** 8,13 ****
--- 8,14 ----
              xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
    <include name="Gio" version="2.0"/>
    <package name="ostree-1"/>
+   <c:include name="ostree.h"/>
    <namespace name="OSTree"
               version="1.0"
               shared-library="libostree-1.so.1"
***************
*** 520,595 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="BOOTLOADER"
-                     c:identifier="OSTREE_BOOTLOADER"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                        line="26"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_ABOOT"
-                     c:identifier="OSTREE_BOOTLOADER_ABOOT"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-aboot.h"
-                        line="25"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_GET_IFACE"
-                     c:identifier="OSTREE_BOOTLOADER_GET_IFACE"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                        line="28"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_GRUB2"
-                     c:identifier="OSTREE_BOOTLOADER_GRUB2"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-grub2.h"
-                        line="25"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_SYSLINUX"
-                     c:identifier="OSTREE_BOOTLOADER_SYSLINUX"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-syslinux.h"
-                        line="25"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_UBOOT"
-                     c:identifier="OSTREE_BOOTLOADER_UBOOT"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-uboot.h"
-                        line="29"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="BOOTLOADER_ZIPL"
-                     c:identifier="OSTREE_BOOTLOADER_ZIPL"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-zipl.h"
-                        line="25"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
      <class name="BootconfigParser"
             c:symbol-prefix="bootconfig_parser"
             c:type="OstreeBootconfigParser"
--- 521,526 ----
***************
*** 824,991 ****
          </parameters>
        </method>
      </class>
-     <record name="Bootloader" c:type="OstreeBootloader" disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                        line="30"/>
-     </record>
-     <record name="BootloaderAboot"
-             c:type="OstreeBootloaderAboot"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-aboot.h"
-                        line="28"/>
-     </record>
-     <record name="BootloaderGrub2"
-             c:type="OstreeBootloaderGrub2"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-grub2.h"
-                        line="28"/>
-     </record>
-     <record name="BootloaderInterface" c:type="OstreeBootloaderInterface">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                        line="53"/>
-       <field name="g_iface" writable="1">
-         <type name="GObject.TypeInterface" c:type="GTypeInterface"/>
-       </field>
-       <field name="query">
-         <callback name="query" throws="1">
-           <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                            line="38"/>
-           <return-value transfer-ownership="none">
-             <type name="gboolean" c:type="gboolean"/>
-           </return-value>
-           <parameters>
-             <parameter name="bootloader" transfer-ownership="none">
-               <type name="Bootloader" c:type="OstreeBootloader*"/>
-             </parameter>
-             <parameter name="out_is_active" transfer-ownership="none">
-               <type name="gboolean" c:type="gboolean*"/>
-             </parameter>
-             <parameter name="cancellable" transfer-ownership="none">
-               <type name="Gio.Cancellable" c:type="GCancellable*"/>
-             </parameter>
-           </parameters>
-         </callback>
-       </field>
-       <field name="get_name">
-         <callback name="get_name">
-           <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                            line="42"/>
-           <return-value transfer-ownership="none">
-             <type name="utf8" c:type="const char*"/>
-           </return-value>
-           <parameters>
-             <parameter name="self" transfer-ownership="none">
-               <type name="Bootloader" c:type="OstreeBootloader*"/>
-             </parameter>
-           </parameters>
-         </callback>
-       </field>
-       <field name="write_config" introspectable="0">
-         <callback name="write_config" introspectable="0" throws="1">
-           <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                            line="43"/>
-           <return-value transfer-ownership="none">
-             <type name="gboolean" c:type="gboolean"/>
-           </return-value>
-           <parameters>
-             <parameter name="self" transfer-ownership="none">
-               <type name="Bootloader" c:type="OstreeBootloader*"/>
-             </parameter>
-             <parameter name="bootversion" transfer-ownership="none">
-               <type name="gint" c:type="int"/>
-             </parameter>
-             <parameter name="new_deployments" transfer-ownership="none">
-               <array name="GLib.PtrArray" c:type="GPtrArray*">
-                 <type name="gpointer" c:type="gpointer"/>
-               </array>
-             </parameter>
-             <parameter name="cancellable" transfer-ownership="none">
-               <type name="Gio.Cancellable" c:type="GCancellable*"/>
-             </parameter>
-           </parameters>
-         </callback>
-       </field>
-       <field name="post_bls_sync">
-         <callback name="post_bls_sync" throws="1">
-           <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                            line="48"/>
-           <return-value transfer-ownership="none">
-             <type name="gboolean" c:type="gboolean"/>
-           </return-value>
-           <parameters>
-             <parameter name="self" transfer-ownership="none">
-               <type name="Bootloader" c:type="OstreeBootloader*"/>
-             </parameter>
-             <parameter name="bootversion" transfer-ownership="none">
-               <type name="gint" c:type="int"/>
-             </parameter>
-             <parameter name="cancellable" transfer-ownership="none">
-               <type name="Gio.Cancellable" c:type="GCancellable*"/>
-             </parameter>
-           </parameters>
-         </callback>
-       </field>
-       <field name="is_atomic">
-         <callback name="is_atomic">
-           <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                            line="52"/>
-           <return-value transfer-ownership="none">
-             <type name="gboolean" c:type="gboolean"/>
-           </return-value>
-           <parameters>
-             <parameter name="self" transfer-ownership="none">
-               <type name="Bootloader" c:type="OstreeBootloader*"/>
-             </parameter>
-           </parameters>
-         </callback>
-       </field>
-     </record>
-     <record name="BootloaderSyslinux"
-             c:type="OstreeBootloaderSyslinux"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-syslinux.h"
-                        line="28"/>
-     </record>
-     <record name="BootloaderUboot"
-             c:type="OstreeBootloaderUboot"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-uboot.h"
-                        line="32"/>
-     </record>
-     <record name="BootloaderZipl" c:type="OstreeBootloaderZipl" disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-zipl.h"
-                        line="28"/>
-     </record>
-     <function-macro name="CHECKSUM_INPUT_STREAM"
-                     c:identifier="OSTREE_CHECKSUM_INPUT_STREAM"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="28"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="CHECKSUM_INPUT_STREAM_CLASS"
-                     c:identifier="OSTREE_CHECKSUM_INPUT_STREAM_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="29"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="CHECKSUM_INPUT_STREAM_GET_CLASS"
-                     c:identifier="OSTREE_CHECKSUM_INPUT_STREAM_GET_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="32"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="CHECK_VERSION"
                      c:identifier="OSTREE_CHECK_VERSION"
                      version="2017.4"
--- 755,760 ----
***************
*** 1152,1257 ****
     values, for bare-user-only mode. (Since: 2021.4.)</doc>
        </member>
      </bitfield>
-     <class name="ChecksumInputStream"
-            c:symbol-prefix="checksum_input_stream"
-            c:type="OstreeChecksumInputStream"
-            parent="Gio.FilterInputStream"
-            glib:type-name="OstreeChecksumInputStream"
-            glib:get-type="ostree_checksum_input_stream_get_type"
-            glib:type-struct="ChecksumInputStreamClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="57"/>
-       <constructor name="new" c:identifier="ostree_checksum_input_stream_new">
-         <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                          line="63"/>
-         <return-value transfer-ownership="full">
-           <type name="ChecksumInputStream"
-                 c:type="OstreeChecksumInputStream*"/>
-         </return-value>
-         <parameters>
-           <parameter name="stream" transfer-ownership="none">
-             <type name="Gio.InputStream" c:type="GInputStream*"/>
-           </parameter>
-           <parameter name="checksum" transfer-ownership="none">
-             <type name="GLib.Checksum" c:type="GChecksum*"/>
-           </parameter>
-         </parameters>
-       </constructor>
-       <property name="checksum"
-                 writable="1"
-                 construct-only="1"
-                 transfer-ownership="none">
-         <type name="gpointer" c:type="gpointer"/>
-       </property>
-       <field name="parent_instance">
-         <type name="Gio.FilterInputStream" c:type="GFilterInputStream"/>
-       </field>
-       <field name="priv" readable="0" private="1">
-         <type name="ChecksumInputStreamPrivate"
-               c:type="OstreeChecksumInputStreamPrivate*"/>
-       </field>
-     </class>
-     <record name="ChecksumInputStreamClass"
-             c:type="OstreeChecksumInputStreamClass"
-             glib:is-gtype-struct-for="ChecksumInputStream">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="57"/>
-       <field name="parent_class">
-         <type name="Gio.FilterInputStreamClass"
-               c:type="GFilterInputStreamClass"/>
-       </field>
-       <field name="_g_reserved1" introspectable="0">
-         <callback name="_g_reserved1">
-           <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                            line="52"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved2" introspectable="0">
-         <callback name="_g_reserved2">
-           <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                            line="53"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved3" introspectable="0">
-         <callback name="_g_reserved3">
-           <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                            line="54"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved4" introspectable="0">
-         <callback name="_g_reserved4">
-           <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                            line="55"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved5" introspectable="0">
-         <callback name="_g_reserved5">
-           <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                            line="56"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-     </record>
-     <record name="ChecksumInputStreamPrivate"
-             c:type="OstreeChecksumInputStreamPrivate"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="36"/>
-     </record>
      <record name="CollectionRef"
              c:type="OstreeCollectionRef"
              version="2018.6"
--- 921,926 ----
***************
*** 2271,2286 ****
                         line="1481"/>
        <type name="utf8" c:type="gchar*"/>
      </constant>
-     <function-macro name="GPG_VERIFIER"
-                     c:identifier="OSTREE_GPG_VERIFIER"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-gpg-verifier.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="obj">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="GPG_VERIFY_RESULT"
                      c:identifier="OSTREE_GPG_VERIFY_RESULT"
                      introspectable="0">
--- 1940,1945 ----
***************
*** 2487,2496 ****
               line="127">Use the default output format</doc>
        </member>
      </bitfield>
-     <record name="GpgVerifier" c:type="OstreeGpgVerifier" disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-gpg-verifier.h"
-                        line="35"/>
-     </record>
      <class name="GpgVerifyResult"
             c:symbol-prefix="gpg_verify_result"
             c:type="OstreeGpgVerifyResult"
--- 2146,2151 ----
***************
*** 2845,2930 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="IS_BOOTLOADER"
-                     c:identifier="OSTREE_IS_BOOTLOADER"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader.h"
-                        line="27"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_BOOTLOADER_ABOOT"
-                     c:identifier="OSTREE_IS_BOOTLOADER_ABOOT"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-aboot.h"
-                        line="26"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_BOOTLOADER_GRUB2"
-                     c:identifier="OSTREE_IS_BOOTLOADER_GRUB2"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-grub2.h"
-                        line="26"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_BOOTLOADER_SYSLINUX"
-                     c:identifier="OSTREE_IS_BOOTLOADER_SYSLINUX"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-syslinux.h"
-                        line="26"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_BOOTLOADER_UBOOT"
-                     c:identifier="OSTREE_IS_BOOTLOADER_UBOOT"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-uboot.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_BOOTLOADER_ZIPL"
-                     c:identifier="OSTREE_IS_BOOTLOADER_ZIPL"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-bootloader-zipl.h"
-                        line="26"/>
-       <parameters>
-         <parameter name="inst">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_CHECKSUM_INPUT_STREAM"
-                     c:identifier="OSTREE_IS_CHECKSUM_INPUT_STREAM"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_CHECKSUM_INPUT_STREAM_CLASS"
-                     c:identifier="OSTREE_IS_CHECKSUM_INPUT_STREAM_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-checksum-input-stream.h"
-                        line="31"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="IS_DEPLOYMENT"
                      c:identifier="OSTREE_IS_DEPLOYMENT"
                      introspectable="0">
--- 2500,2505 ----
***************
*** 2935,2950 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="IS_GPG_VERIFIER"
-                     c:identifier="OSTREE_IS_GPG_VERIFIER"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-gpg-verifier.h"
-                        line="32"/>
-       <parameters>
-         <parameter name="obj">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="IS_GPG_VERIFY_RESULT"
                      c:identifier="OSTREE_IS_GPG_VERIFY_RESULT"
                      introspectable="0">
--- 2510,2515 ----
***************
*** 2955,3020 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="IS_LIBARCHIVE_INPUT_STREAM"
-                     c:identifier="OSTREE_IS_LIBARCHIVE_INPUT_STREAM"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="32"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_LIBARCHIVE_INPUT_STREAM_CLASS"
-                     c:identifier="OSTREE_IS_LIBARCHIVE_INPUT_STREAM_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="33"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_LZMA_COMPRESSOR"
-                     c:identifier="OSTREE_IS_LZMA_COMPRESSOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="29"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_LZMA_COMPRESSOR_CLASS"
-                     c:identifier="OSTREE_IS_LZMA_COMPRESSOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_LZMA_DECOMPRESSOR"
-                     c:identifier="OSTREE_IS_LZMA_DECOMPRESSOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="29"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_LZMA_DECOMPRESSOR_CLASS"
-                     c:identifier="OSTREE_IS_LZMA_DECOMPRESSOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="IS_MUTABLE_TREE"
                      c:identifier="OSTREE_IS_MUTABLE_TREE"
                      introspectable="0">
--- 2520,2525 ----
***************
*** 3065,3090 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="IS_REPO_FILE_ENUMERATOR"
-                     c:identifier="OSTREE_IS_REPO_FILE_ENUMERATOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="31"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="IS_REPO_FILE_ENUMERATOR_CLASS"
-                     c:identifier="OSTREE_IS_REPO_FILE_ENUMERATOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="32"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="IS_SEPOLICY"
                      c:identifier="OSTREE_IS_SEPOLICY"
                      introspectable="0">
--- 2570,2575 ----
***************
*** 3741,3933 ****
        <source-position filename="Projects/ostree/src/libostree/ostree-kernel-args.h"
                         line="28"/>
      </record>
-     <function-macro name="LIBARCHIVE_INPUT_STREAM"
-                     c:identifier="OSTREE_LIBARCHIVE_INPUT_STREAM"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LIBARCHIVE_INPUT_STREAM_CLASS"
-                     c:identifier="OSTREE_LIBARCHIVE_INPUT_STREAM_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="31"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LIBARCHIVE_INPUT_STREAM_GET_CLASS"
-                     c:identifier="OSTREE_LIBARCHIVE_INPUT_STREAM_GET_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="34"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_COMPRESSOR"
-                     c:identifier="OSTREE_LZMA_COMPRESSOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="27"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_COMPRESSOR_CLASS"
-                     c:identifier="OSTREE_LZMA_COMPRESSOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="28"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_COMPRESSOR_GET_CLASS"
-                     c:identifier="OSTREE_LZMA_COMPRESSOR_GET_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="31"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_DECOMPRESSOR"
-                     c:identifier="OSTREE_LZMA_DECOMPRESSOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="27"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_DECOMPRESSOR_CLASS"
-                     c:identifier="OSTREE_LZMA_DECOMPRESSOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="28"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="LZMA_DECOMPRESSOR_GET_CLASS"
-                     c:identifier="OSTREE_LZMA_DECOMPRESSOR_GET_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="31"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <record name="LibarchiveInputStream" c:type="OstreeLibarchiveInputStream">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="46"/>
-       <field name="parent_instance" writable="1">
-         <type name="Gio.InputStream" c:type="GInputStream"/>
-       </field>
-       <field name="priv" readable="0" private="1">
-         <type name="LibarchiveInputStreamPrivate"
-               c:type="OstreeLibarchiveInputStreamPrivate*"/>
-       </field>
-     </record>
-     <record name="LibarchiveInputStreamClass"
-             c:type="OstreeLibarchiveInputStreamClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="59"/>
-       <field name="parent_class" writable="1">
-         <type name="Gio.InputStreamClass" c:type="GInputStreamClass"/>
-       </field>
-       <field name="_g_reserved1" introspectable="0">
-         <callback name="_g_reserved1">
-           <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                            line="54"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved2" introspectable="0">
-         <callback name="_g_reserved2">
-           <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                            line="55"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved3" introspectable="0">
-         <callback name="_g_reserved3">
-           <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                            line="56"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved4" introspectable="0">
-         <callback name="_g_reserved4">
-           <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                            line="57"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-       <field name="_g_reserved5" introspectable="0">
-         <callback name="_g_reserved5">
-           <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                            line="58"/>
-           <return-value transfer-ownership="none">
-             <type name="none" c:type="void"/>
-           </return-value>
-         </callback>
-       </field>
-     </record>
-     <record name="LibarchiveInputStreamPrivate"
-             c:type="OstreeLibarchiveInputStreamPrivate"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-libarchive-input-stream.h"
-                        line="38"/>
-     </record>
-     <record name="LzmaCompressor" c:type="OstreeLzmaCompressor" disguised="1">
-       <doc xml:space="preserve"
-            filename="Projects/ostree/src/libostree/ostree-lzma-compressor.c"
-            line="45">Zlib decompression</doc>
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="34"/>
-     </record>
-     <record name="LzmaCompressorClass" c:type="OstreeLzmaCompressorClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-compressor.h"
-                        line="39"/>
-       <field name="parent_class" writable="1">
-         <type name="GObject.ObjectClass" c:type="GObjectClass"/>
-       </field>
-     </record>
-     <record name="LzmaDecompressor"
-             c:type="OstreeLzmaDecompressor"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="34"/>
-     </record>
-     <record name="LzmaDecompressorClass" c:type="OstreeLzmaDecompressorClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-lzma-decompressor.h"
-                        line="39"/>
-       <field name="parent_class" writable="1">
-         <type name="GObject.ObjectClass" c:type="GObjectClass"/>
-       </field>
-     </record>
      <constant name="MAX_METADATA_SIZE"
                value="10485760"
                c:type="OSTREE_MAX_METADATA_SIZE">
--- 3226,3231 ----
***************
*** 4661,4696 ****
          </parameter>
        </parameters>
      </function-macro>
-     <function-macro name="REPO_FILE_ENUMERATOR"
-                     c:identifier="OSTREE_REPO_FILE_ENUMERATOR"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="29"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="REPO_FILE_ENUMERATOR_CLASS"
-                     c:identifier="OSTREE_REPO_FILE_ENUMERATOR_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="30"/>
-       <parameters>
-         <parameter name="k">
-         </parameter>
-       </parameters>
-     </function-macro>
-     <function-macro name="REPO_FILE_ENUMERATOR_GET_CLASS"
-                     c:identifier="OSTREE_REPO_FILE_ENUMERATOR_GET_CLASS"
-                     introspectable="0">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="33"/>
-       <parameters>
-         <parameter name="o">
-         </parameter>
-       </parameters>
-     </function-macro>
      <function-macro name="REPO_FILE_GET_CLASS"
                      c:identifier="OSTREE_REPO_FILE_GET_CLASS"
                      introspectable="0">
--- 3959,3964 ----
***************
*** 13536,13555 ****
          <type name="GObject.ObjectClass" c:type="GObjectClass"/>
        </field>
      </record>
-     <record name="RepoFileEnumerator"
-             c:type="OstreeRepoFileEnumerator"
-             disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="35"/>
-     </record>
-     <record name="RepoFileEnumeratorClass"
-             c:type="OstreeRepoFileEnumeratorClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-repo-file-enumerator.h"
-                        line="41"/>
-       <field name="parent_class" writable="1">
-         <type name="Gio.FileEnumeratorClass" c:type="GFileEnumeratorClass"/>
-       </field>
-     </record>
      <interface name="RepoFinder"
                 c:symbol-prefix="repo_finder"
                 c:type="OstreeRepoFinder"
--- 12804,12809 ----
***************
*** 15017,15055 ****
               line="1563">Skip all other signature verification methods</doc>
        </member>
      </bitfield>
-     <record name="RollsumMatches" c:type="OstreeRollsumMatches">
-       <source-position filename="Projects/ostree/src/libostree/ostree-rollsum.h"
-                        line="35"/>
-       <field name="from_rollsums" writable="1">
-         <type name="GLib.HashTable" c:type="GHashTable*">
-           <type name="gpointer" c:type="gpointer"/>
-           <type name="gpointer" c:type="gpointer"/>
-         </type>
-       </field>
-       <field name="to_rollsums" writable="1">
-         <type name="GLib.HashTable" c:type="GHashTable*">
-           <type name="gpointer" c:type="gpointer"/>
-           <type name="gpointer" c:type="gpointer"/>
-         </type>
-       </field>
-       <field name="crcmatches" writable="1">
-         <type name="guint" c:type="guint"/>
-       </field>
-       <field name="bufmatches" writable="1">
-         <type name="guint" c:type="guint"/>
-       </field>
-       <field name="total" writable="1">
-         <type name="guint" c:type="guint"/>
-       </field>
-       <field name="match_size" writable="1">
-         <type name="guint64" c:type="guint64"/>
-       </field>
-       <field name="matches" writable="1">
-         <array name="GLib.PtrArray" c:type="GPtrArray*">
-           <type name="gpointer" c:type="gpointer"/>
-         </array>
-       </field>
-     </record>
      <function-macro name="SEPOLICY"
                      c:identifier="OSTREE_SEPOLICY"
                      introspectable="0">
--- 14271,14276 ----
***************
*** 16155,16298 ****
            </parameter>
          </parameters>
        </method>
-       <method name="dummy_add_pk"
-               c:identifier="ostree_sign_dummy_add_pk"
-               throws="1">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="72"/>
-         <return-value transfer-ownership="none">
-           <type name="gboolean" c:type="gboolean"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-           <parameter name="key" transfer-ownership="none">
-             <type name="GLib.Variant" c:type="GVariant*"/>
-           </parameter>
-         </parameters>
-       </method>
-       <method name="dummy_data"
-               c:identifier="ostree_sign_dummy_data"
-               throws="1">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="55"/>
-         <return-value transfer-ownership="none">
-           <type name="gboolean" c:type="gboolean"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-           <parameter name="data" transfer-ownership="none">
-             <type name="GLib.Bytes" c:type="GBytes*"/>
-           </parameter>
-           <parameter name="signature" transfer-ownership="none">
-             <type name="GLib.Bytes" c:type="GBytes**"/>
-           </parameter>
-           <parameter name="cancellable"
-                      transfer-ownership="none"
-                      nullable="1"
-                      allow-none="1">
-             <type name="Gio.Cancellable" c:type="GCancellable*"/>
-           </parameter>
-         </parameters>
-       </method>
-       <method name="dummy_data_verify"
-               c:identifier="ostree_sign_dummy_data_verify"
-               throws="1">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="61"/>
-         <return-value transfer-ownership="none">
-           <type name="gboolean" c:type="gboolean"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-           <parameter name="data" transfer-ownership="none">
-             <type name="GLib.Bytes" c:type="GBytes*"/>
-           </parameter>
-           <parameter name="signatures" transfer-ownership="none">
-             <type name="GLib.Variant" c:type="GVariant*"/>
-           </parameter>
-           <parameter name="success_message" transfer-ownership="none">
-             <type name="utf8" c:type="char**"/>
-           </parameter>
-         </parameters>
-       </method>
-       <method name="dummy_get_name" c:identifier="ostree_sign_dummy_get_name">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="53"/>
-         <return-value transfer-ownership="none">
-           <type name="utf8" c:type="const gchar*"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-         </parameters>
-       </method>
-       <method name="dummy_metadata_format"
-               c:identifier="ostree_sign_dummy_metadata_format">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="68"/>
-         <return-value transfer-ownership="none">
-           <type name="utf8" c:type="const gchar*"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-         </parameters>
-       </method>
-       <method name="dummy_metadata_key"
-               c:identifier="ostree_sign_dummy_metadata_key">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="67"/>
-         <return-value transfer-ownership="none">
-           <type name="utf8" c:type="const gchar*"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-         </parameters>
-       </method>
-       <method name="dummy_set_pk"
-               c:identifier="ostree_sign_dummy_set_pk"
-               throws="1">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="71"/>
-         <return-value transfer-ownership="none">
-           <type name="gboolean" c:type="gboolean"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-           <parameter name="key" transfer-ownership="none">
-             <type name="GLib.Variant" c:type="GVariant*"/>
-           </parameter>
-         </parameters>
-       </method>
-       <method name="dummy_set_sk"
-               c:identifier="ostree_sign_dummy_set_sk"
-               throws="1">
-         <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                          line="70"/>
-         <return-value transfer-ownership="none">
-           <type name="gboolean" c:type="gboolean"/>
-         </return-value>
-         <parameters>
-           <instance-parameter name="self" transfer-ownership="none">
-             <type name="Sign" c:type="OstreeSign*"/>
-           </instance-parameter>
-           <parameter name="key" transfer-ownership="none">
-             <type name="GLib.Variant" c:type="GVariant*"/>
-           </parameter>
-         </parameters>
-       </method>
        <method name="ed25519_add_pk"
                c:identifier="ostree_sign_ed25519_add_pk"
                throws="1">
--- 15376,15381 ----
***************
*** 16692,16708 ****
          </parameters>
        </method>
      </interface>
-     <record name="SignDummy" c:type="OstreeSignDummy" disguised="1">
-       <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                        line="36"/>
-     </record>
-     <record name="SignDummyClass" c:type="OstreeSignDummyClass">
-       <source-position filename="Projects/ostree/src/libostree/ostree-sign-dummy.h"
-                        line="37"/>
-       <field name="parent_class" writable="1">
-         <type name="GObject.ObjectClass" c:type="GObjectClass"/>
-       </field>
-     </record>
      <record name="SignEd25519" c:type="OstreeSignEd25519" disguised="1">
        <source-position filename="Projects/ostree/src/libostree/ostree-sign-ed25519.h"
                         line="36"/>
--- 15775,15780 ----
```

</p>
</details>

Reply:
I've bumped the GObject Introspection version to 1.51.5

Reply:
Interesting, that certainly strips out several headers that are internal but not named `*-private.h`.

I think this is good, but it's technically an IABI break since it's removing things. @cgwalters what do you think?